### ISSUE TITLE
Lock api support?

### ISSUE THREAD

**Issue by JamesYFC** [jasonwhite/rudolfs#25]
Does this repo support lfs file locking?

When I try to run `lfs lock` I get errors.
![image](https://user-images.githubusercontent.com/16425226/106992306-d6811700-67c3-11eb-9b1f-2ebddeccb740.png)

**Comment by jasonwhite** [comment]
No, file locking is not yet supported. PRs are welcome if anyone wants to take a stab at it.

**Comment by Dmitry1987** [comment]
latest version says "Remote "origin" does not support the LFS locking API. Consider disabling it with"  😢    
what is the best LFS local server you guys are using? i need to host it on ec2 for all internal large files storage. I saw gitlab and Nexus are an option, but not sure whether it's an overkill to maintain one of them just for the LFS storage 😅  was happy I found this solution, seemed up to date and says 'high performance' 💡  I had big hopes.  btw it does not work, getting stuck on upload and 0%, strange.

**Comment by jasonwhite** [comment]
@Dmitry1987 File locking support should be an optional feature with the Git LFS client. Without server support, commands like `git lfs lock <path>` won't work. I've never needed to use file locking with Git LFS and I'm dubious of its utility, but I'm open to being convinced otherwise. I have used very old version control systems with file locking and it has always been a pain.

Sounds like the upload getting stuck is a separate issue. If you want help solving it, please create a new issue with any error messages.

> what is the best LFS local server you guys are using?

As far as I'm aware, this is still the best Git LFS server out there in terms of performance and stability. GitLab and Nexus might be better options if you're already into those ecosystems. I've used both some years ago and found them to have poor performance, but they may have improved since then. They never got close to saturating disk/network throughput in my tests. They also lacked some features that I wanted, like encryption and caching. YMMV.

**Comment by Dmitry1987** [comment]
thanks for your inputs @jasonwhite 🙏 
in our case the artists are the ones locking files (3d, 2d, anything binary that's in process of being modified, only one person should work at a time, otherwise will be a mess). so the locking acts mostly like signaling mechanism so everyone else will know that a file is WIP and there's no point modifying it at once since either side of the work will get lost (either the source gets discarded as a whole or the destination, and it's waste of someone's time for the artwork / sculpting / animation / etc').
I don't think there's a use case for locking in text-files world, code can get merged and diffs are useful to know 'what was planned to be refactored' even if it gets discarded, so I agree with you on that part. so far it's been useful only for art teams. 
We decided to use gitlab after some discussions, it supports locking and also we might use the CI runners and pipelines. 
thanks! 👋
